### PR TITLE
Impersonation fix

### PR DESCRIPTION
**Pull request by sean-brydon**
Fixes impersoantion 

MoreInfo: 
https://twist.com/a/195135/msg/1414863/m/73569232
https://twist.com/a/195135/msg/1414863/m/73569629

### CODE REVIEW

File: packages/features/ee/impersonation/lib/ImpersonationProvider.ts [hunk 1]
```diff
@@ -35,6 +38,8 @@ const auditAndReturnNextUser = async (
     impersonatedByUID,
   };
 
+  console.log(obj);
```
Comment (zomars) [inline packages/features/ee/impersonation/lib/ImpersonationProvider.ts:41]:
```suggestion
```

File: packages/features/ee/impersonation/lib/ImpersonationProvider.ts [hunk 2]
```diff
@@ -50,7 +55,7 @@ const ImpersonationProvider = CredentialsProvider({
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore need to figure out how to correctly type this
     const session = await getSession({ req });
-    const teamId = creds?.teamId ? asNumberOrThrow(creds.teamId) : undefined;
+    const teamId = creds?.teamId ? teamIdschema.parse(creds?.teamId).teamId : undefined;
```
Comment (zomars) [inline packages/features/ee/impersonation/lib/ImpersonationProvider.ts:58]:
```suggestion
    const teamId = creds?.teamId ? teamIdschema.parse(creds).teamId : undefined;
```
???

**Review by zomars — COMMENTED** [review]
The zod schema is a bit confusing

File: packages/features/ee/impersonation/lib/ImpersonationProvider.ts (hunk 2 above)
Comment (sean-brydon) [reply packages/features/ee/impersonation/lib/ImpersonationProvider.ts:57]: Fixing

File: packages/features/ee/impersonation/lib/ImpersonationProvider.ts [hunk 3]
```diff
@@ -103,35 +107,39 @@ const ImpersonationProvider = CredentialsProvider({
...
-        id: session?.user.id,
-      },
-      include: {
-        teams: {
-          where: {
-            AND: [
-              {
-                role: {
-                  in: ["ADMIN", "OWNER"],
+    if (teamId) {
```
Comment (zomars) [inline packages/features/ee/impersonation/lib/ImpersonationProvider.ts:110]: [Prefer Early Returns](https://handbook.cal.com/engineering/best-practices/prefer-early-returns) 

```suggestion
    if (!teamId) throw new Error("You do not have permission to do this.");
```

**Review by zomars — APPROVED** [review]

**Pull request merged**